docs/observability/index.mdx

```diff
@@ -0,0 +1,301 @@
+
+The first thing you might want to trace is all your OpenAI calls. LangSmith makes this easy with the `wrap_openai/wrapOpenAI` wrappers.
+All you have to do is modify your code to use the wrapped client instead of using the `OpenAI` client directly.
+
+<CodeTabs
+  groupId="client-language"
+  tabs={[
+    python({ label: "Python" })`
+      from openai import OpenAI
+      # highlight-next-line
+      from langsmith.wrappers import wrap_openai
+      
+      # highlight-next-line
+      openai_client = wrap_openai(OpenAI())
+      
+      # This is the retriever we will use in RAG
+      # This is mocked out, but it could be anything we want
+      def retriever(query: str):
+          results = ["Harrison worked at Kensho"]
+          return results
+      
+      # This is the end-to-end RAG chain.
+      # It does a retrieval step then calls OpenAI
+      def rag(question):
+          docs = retriever(question)
+          system_message = """Answer the users question using only the provided information below:
+          
+          {docs}""".format(docs="\\n".join(docs))
+          
+          return openai_client.chat.completions.create(
+              messages=[
+                  {"role": "system", "content": system_message},
+                  {"role": "user", "content": question},
+              ],
+              model="gpt-4o-mini",
+          )
+    `,
+    typescript({ label: "TypeScript" })`
+      import { OpenAI } from "openai";
+      // highlight-next-line
+      import { wrapOpenAI } from "langsmith/wrappers";
+      
+      // highlight-next-line
+      const openAIClient = wrapOpenAI(new OpenAI());
+      
+      // This is the retriever we will use in RAG
+      // This is mocked out, but it could be anything we want
+      async function retriever(query: string) {
+        return ["This is a document"];
+      }
+      
+      // This is the end-to-end RAG chain.
+      // It does a retrieval step then calls OpenAI
+      async function rag(question: string) {
+        const docs = await retriever(question);
+        
+        const systemMessage =
+          "Answer the users question using only the provided information below:\\n\\n" +
+          docs.join("\\n");
+          
+        return await openAIClient.chat.completions.create({
+          messages: [
+            { role: "system", content: systemMessage },
+            { role: "user", content: question },
+          ],
+          model: "gpt-4o-mini",
+        });
+      }
+    `,
+  ]}
+/>
+
+Now when you call your application as follows:
+
+```python
+rag("where did harrison work")
+```
+
+This will produce a trace of just the OpenAI call - it should look something like [this](https://smith.langchain.com/public/e7b7d256-10fe-4d49-a8d5-36ca8e5af0d2/r)
+
+![](./tutorials/static/tracing_tutorial_openai.png)
+
+## 6. Trace entire application
+
+You can also use the `traceable` decorator to trace your entire application instead of just the LLM calls.
```

Comment:
add docs links

https://langsmith-docs-bdk0fivr6-langchain.vercel.app/reference/js/functions/traceable.traceable (why does the URL change)
https://docs.smith.langchain.com/reference/python/run_helpers/langsmith.run_helpers.traceable